src/formatter.rs

```diff
@@ -65,25 +77,18 @@ where
                 .or_else(|| ctx.lookup_current());
 
             if let Some(span) = span {
-                let span_path = span.scope().from_root().map(|span| span.name()).fold(
-                    String::new(),
-                    |mut acc, name| {
-                        let add_separator = !acc.is_empty();
-
-                        if add_separator {
-                            acc.reserve(name.len() + 1);
-                            acc.push('>');
-                        } else {
-                            acc.reserve(name.len());
-                        }
-
-                        acc.push_str(name);
-                        acc
-                    },
-                );
-
                 serializer.serialize_entry("span", span.name())?;
-                serializer.serialize_entry("span_path", &span_path)?;
+                serializer.serialize_key("span_path")?;
+                serializer.writer.write_char('=')?;
+                let mut insert_sep = false;
+                for span in span.scope().from_root() {
```

Comment:
If I read this code correctly it doesn't escape the span_path?

Span names can contain spaces (and maybe also quotes etc? not sure), and if they do the span_path needs to be escaped.

I missed adding a test case for that.

Reply:
ah I see hm yeah thats unfortunate 🤔 I'll give it a shot and see what I can come up with

Reply:
we should probably also make those extra fields configurable, so they can be turned off for projects with high log volumes. I'll create an issue on that

Reply:
Alright I've tweaked it now:

- It first goes through the span names and checks if any of them needs quoting
- If so it precomputes the necessary capacity and writes things to an intermediate buffer and then writes that buffer with quoting
- If not then it just writes the span names directly

Now my benchmark makes 900.014 with a quoted span and the same 700.014 otherwise.

You could probably save the intermediate buffer all together by like double quoting the span names as you're writing it or something. Left as an exercise for the reader :)

Reply:
Nice!